protocol/auditlog.go

```diff
@@ -112,26 +129,28 @@ func (l ConiksAuditLog) Insert(addr string, signKey sign.PublicKey,
 
 		// verify the consistency of each new STR before inserting
 		// into the audit log
-		if err := verifySTRConsistency(signKey, h.latestSTR, str); err != nil {
+		if err := h.verifySTRConsistency(snaps[i-1], str); err != nil {
 			return err
 		}
-
-		h.updateLatestSTR(snaps[i])
 	}
 
 	// Finally, add the new history to the log
+	if len(snaps) > 1 {
```

Comment:
Also, this path is already covered in tests that include https://github.com/coniks-sys/coniks-go/blob/generic-auditor/protocol/auditlog_test.go#L29 via https://github.com/coniks-sys/coniks-go/blob/generic-auditor/protocol/testutil.go#L51